04_numeronym/brvishnu92/main_test.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+//main() test
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+	main()
+	expected := `[a11y K8s abc]
+`
+	actual := buf.String()
+	assert.Equalf(t, expected, actual, "got %v want %v", actual, expected)
+}
+
+func TestNumeronyms(t *testing.T) {
+	testCases := map[string]struct {
+		input    []string
+		expected []string
+	}{
+		"test1":  {input: []string{"demon", "GolangCli", "p"}, expected: []string{"d3n", "G7i", "p"}},
+		"empty":  {input: []string{}, expected: []string{}},
+		"oneval": {input: []string{"a"}, expected: []string{"a"}},
+	}
```

Comment:
You should have some tests that test boundary conditions. For example, you have the conditional `x > 3` in your code, so you should have a test string of length 3 and one of length 4 to ensure the condition is operating correctly. Sometimes you realise a test should be `>=` instead of just `>` and boundary condition tests often pick those up.